src/main/resources/test_files/runtime-spark/FunctionUserDefinedDynamic/NamedFunctionRefRDD3.jq

```diff
@@ -0,0 +1,5 @@
+(:JIQS: ShouldCrash; ErrorCode="SENR0001"; :)
+declare function fn1() {3};
+parallelize(fn1#0)
```

Comment:
This would also be worth an issue (but I don't expect this to be addressed in the next couple of months as this is advanced)

Reply:
This was actually by design on my part at the time. XQuery data model defines functions as 'Functions have no identity, cannot be compared, and have no serialization.' at https://www.w3.org/TR/xpath-datamodel-30/#dt-function-item

Do we prefer to have some sort of serialization for functions? If so, I will create an issue

Reply:
Good point. We do not need to serialize functions -- we can simply follow the spec and throw an error code as specified for unserializable items if it is attempted to serialize a function item -- which this tests apparently already checks. You may also want to throw the appropriate error code if they are compared with value/general comparison.

Reply:
@ghislainfourny The function items will throw a non-atomic comparison error (JNTY0004) in the new implementation. 

I will merge the PR on your approval of the tests & error codes.